wasm.md

```diff
@@ -849,30 +848,36 @@ It could also be declared implicitly when a `TypeUse` is a `TypeDecls`, in this
 
 Function declarations can look quite different depending on which fields are ommitted and what the context is.
 Here, we allow for an "abstract" function declaration using syntax `func_::___`, and a more concrete one which allows arbitrary order of declaration of parameters, locals, and results.
+Functions can either be specified by giving a type, what locals it allocates, and a function body; or by an import and it's expected type.
```

Comment:
Not sure where to put such an explanation, so I put it under the section `Inline Exports`. We could also deal with all the export desugaring under that section. Let me see how that looks like.

Reply:
Okay, so the problem is that since an empty list of exports is called `.InlineExports`, it needs to be explicitly mentioned in the rules. So if we move this exporting logic to another section, we have this wierd `.InlineExports` thing hanging around.

Reply:
I think I have a solution though that will turn out quite nice.

Reply:
Re: this: now all the exports and imports syntax and rules for desugaring are under their respective sections, so the main part of the semantics actually doesn't deal with them at all. It also becomes more readable I think, because the desugaring rules are almost identical for the different allocated kinds (i.e., functions, globals, etc.)